flake.nix

```diff
@@ -44,5 +50,45 @@
           devShell = devShells.cronosd;
         }
       )
-    );
+    ) // {
+      overlay = final: prev: {
+        buildGoApplication = final.callPackage (import (gomod2nix + "/builder")) {
+          go = final.go_1_17;
+        };
+        bundle-exe = import nix-bundle-exe { pkgs = final; };
+        bundle-exe-tarball = drv:
+          let bundle = final.bundle-exe drv;
+          in
+          final.runCommand bundle.name { } ''
+            "${final.gnutar}/bin/tar" cfzhv $out -C ${bundle} .
+          '';
+      } // (with final;
+        let
+          matrix = lib.cartesianProductOfSets {
+            db_backend = [ "goleveldb" "rocksdb" ];
+            network = [ "mainnet" "testnet" ];
+            relocatable = [ true false ];
+          };
+          binaries = builtins.listToAttrs (builtins.map
```

Comment:
does relocatable matter for goleveldb? I assume it's primarily for RocksDB use cases?
in any case, it's lazy, so not an issue if it's not used

Reply:
In theory goleveldb version has libc dependency at runtime.
I was thinking if we can unify the process with nix, at least for Linux and Mac.